catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/PreferencesSensorMappingTest.java

```diff
@@ -25,45 +25,74 @@
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensor;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
 
-import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertSame;
+import static junit.framework.Assert.assertTrue;
 
-@RunWith(AndroidJUnit4.class)
+@RunWith(Parameterized.class)
 public class PreferencesSensorMappingTest {
 
-	@Test
-	public void testNXTSensorToTextMapping() {
-		Context context = InstrumentationRegistry.getTargetContext().getApplicationContext();
+	private static final int EXPECTED_MAPPING_SIZE = 6;
 
-		final List<Pair<Integer, NXTSensor.Sensor>> correctMapping = new ArrayList<Pair<Integer, NXTSensor.Sensor>>();
+	@Parameterized.Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{"NO_SENSOR", R.string.nxt_no_sensor, NXTSensor.Sensor.NO_SENSOR},
+				{"TOUCH", R.string.nxt_sensor_touch, NXTSensor.Sensor.TOUCH},
+				{"SOUND", R.string.nxt_sensor_sound, NXTSensor.Sensor.SOUND},
+				{"LIGHT_INACTIVE", R.string.nxt_sensor_light, NXTSensor.Sensor.LIGHT_INACTIVE},
+				{"LIGHT_ACTIVE", R.string.nxt_sensor_light_active, NXTSensor.Sensor.LIGHT_ACTIVE},
+				{"ULTRASONIC", R.string.nxt_sensor_ultrasonic, NXTSensor.Sensor.ULTRASONIC},
+		});
+	}
 
-		correctMapping.add(new Pair(R.string.nxt_no_sensor, NXTSensor.Sensor.NO_SENSOR));
-		correctMapping.add(new Pair(R.string.nxt_sensor_touch, NXTSensor.Sensor.TOUCH));
-		correctMapping.add(new Pair(R.string.nxt_sensor_sound, NXTSensor.Sensor.SOUND));
-		correctMapping.add(new Pair(R.string.nxt_sensor_light, NXTSensor.Sensor.LIGHT_INACTIVE));
-		correctMapping.add(new Pair(R.string.nxt_sensor_light_active, NXTSensor.Sensor.LIGHT_ACTIVE));
-		correctMapping.add(new Pair(R.string.nxt_sensor_ultrasonic, NXTSensor.Sensor.ULTRASONIC));
+	@Parameterized.Parameter
+	public String name;
 
-		String[] sensorNames = context.getResources().getStringArray(R.array.nxt_sensor_chooser);
-		String[] sensorPreferencesCodes = NXTSensor.Sensor.getSensorCodes();
+	@Parameterized.Parameter(1)
+	public int sensorNameStringId;
+	@Parameterized.Parameter(2)
+	public NXTSensor.Sensor sensor;
 
-		assertEquals(correctMapping.size(), sensorNames.length);
-		assertEquals(correctMapping.size(), sensorPreferencesCodes.length);
+	private String sensorName;
 
-		for (int i = 0; i < correctMapping.size(); ++i) {
-			assertEquals(context.getString(correctMapping.get(i).first), sensorNames[i]);
+	private ArrayList<String> sensorNames = new ArrayList<>();
+	private ArrayList<String> sensorCodes = new ArrayList<>();
 
-			assertEquals(correctMapping.get(i).second.getSensorCode(), sensorPreferencesCodes[i]);
-		}
+	@Before
+	public void setUp() {
+		Context context = InstrumentationRegistry.getTargetContext().getApplicationContext();
```

Comment:
`...getTargetContext();` should be enough to return the currents instrumented applications contex